components/motor/gpio/motor_encoder.go

```diff
@@ -196,31 +183,32 @@ func (m *EncodedMotor) rpmMonitor(ctx context.Context, goalRPM, goalPos, directi
 			return
 		}
 
-		newPower, err := m.makeAdjustments(ctx, pos, lastPos, goalRPM, goalPos, lastPowerPct, direction, now, lastTime)
-		if err != nil {
+		m.logger.CDebug(ctx, "making adjustments")
+
+		currentRPM := calculateCurrentRpm(pos, lastPos, m.ticksPerRotation, now, lastTime)
+		newPower := calculateNewPower(currentRPM, goalRPM, lastPowerPct, direction, m.rampRate)
+
+		m.mu.Unlock()
+		defer m.mu.Lock()
```

Comment:
I'm a bit confused by this, I'm not seeing where the lock is that requires the unlock and relock

Reply:
I just moved it from above the original setPower to here, I'll check if removing it causes no damage.